code/modules/atmospherics/machinery/airalarm.dm

```diff
@@ -202,17 +209,17 @@
 	if(name == initial(name))
 		name = "[get_area_name(src)] Air Alarm"
 
-	update_icon()
+	power_change()
+	set_frequency(frequency)
 
 /obj/machinery/airalarm/Destroy()
 	SSradio.remove_object(src, frequency)
 	qdel(wires)
 	wires = null
 	return ..()
 
-/obj/machinery/airalarm/Initialize(mapload)
-	. = ..()
-	set_frequency(frequency)
+/obj/machinery/airalarm/on_construction()
+	..(dir,dir)
```

Comment:
`/machinery` doesn't have any code to run for this proc and doesn't take any args. What are you doing here?